tools/PI/DevHome.PI/BarWindowVertical.xaml.cs

```diff
@@ -95,6 +95,16 @@ private void MainPanel_Loaded(object sender, RoutedEventArgs e)
         Width = 70;
     }
 
+    private void ExternalToolsMenu_Opening(object sender, object e)
+    {
+        // Cancel the opening of the menu if there are no items.
+        var flyout = sender as MenuFlyout;
+        if (flyout is not null && flyout.Items is not null && flyout.Items.Count == 0)
+        {
+            flyout.Hide();
```

Comment:
I'm worried users will think this is a bug. "I clicked the button but nothing happened" or "I clicked the button and there was a flash but it closed immediately". Can we disable the button if there's nothing in the menu?

Reply:
There shouldn't be a flash, nor open/close - this event fires before that could happen. Disabling a right-click would have the same effect as this approach, and with the same concern, ie "right click doesn't do anything".

Reply:
I have this fixed in my "external tools" refactor PR, where we won't even have a flyout attached to the button in this case.